LEAF_Request_Portal/sources/System.php

```diff
@@ -686,8 +686,16 @@ public function addAction()
                 ':fillDependency' => $_POST['fillDependency'],
         );
 
-        $this->db->prepared_query('INSERT INTO actions (actionType, actionText, actionTextPasttense, actionIcon, actionAlignment, sort, fillDependency)
-										VALUES (:actionType, :actionText, :actionTextPasttense, :actionIcon, :actionAlignment, :sort, :fillDependency)', $vars);
+        $sql = 'INSERT INTO `actions` (`actionType`, `actionText`,
+                    `actionTextPasttense`, `actionIcon`, `actionAlignment`, `sort`, `fillDependency`)
+                VALUES (:actionType, :actionText, :actionTextPasttense, :actionIcon, :actionAlignment, :sort, :fillDependency)
+                ON DUPLICATE KEY UPDATE `actionText` = :actionText,
+                    `actionTextPasttense` = :actionTextPasttense,
+                    `actionIcon` = :actionIcon,
+                    `actionAlignment` = :actionAlignment, `sort` = :sort,
+                    `fillDependency` = :fillDependency, `deleted` = 0';
+
+        $this->db->prepared_query($sql, $vars);
```

Comment:
I think we might want to check the deleted status and only update an existing one if it's disabled.  The action text (that the table is ordered by) and actionType aren't the same, so users might unintentionally overwrite an existing action if we just post it.

Reply:
Functionality has been added to check for this.